doc/design/draft/named-segments.md

```diff
@@ -0,0 +1,219 @@
+  * Publishers may specify a single segment into which they will publish.
+  * Clients and Servers may specify a single segment into which they send requests and responses, respectively.
+* When no segment is specified, the fallback behavior must be the same as it is currently
+  * Publishers will select the *unique* segment they have write access to.
+  * Clients and Servers will select the *unique* segment they have write access to, for publishing requests and responses, respectively.
+* Creation will fail if a producer requests to use a segment to which it does not have write access.
+
+### Optional additional requirements
+
+* When initializing the runtime, the user may specify via an enum whether they wish to map
+  * All segments to which the current process has read access
+  * All segments to which the current process has write access
+  * All segments to which the current process has either read or write access (deprecated-default)
+  * No segments (future default)
+* The user may additionally map specific segments after initialization by calling
+  * `mapSegmentForReading(name)`
+    * This will check for existence of the specified segment and map it as long as user has read access to it
+    * An informative error will be returned under the following circumstances
+      * The segment has already been mapped
+      * The segment does not exist
+      * The segment exists but the current process does not have read access to it
+  * `mapSegmentForWriting(name)`
+    * This will check for the existence of the specified segment and map it as long as the user has write access to it
+    * An informative error will be returned under the following circumstances
+      * The segment has already been mapped
+      * The segment does not exist
+      * The segment exists but the current process does not have write access to it
+* When the user attempts to create a producer that requests a segment which is not mapped, this will trigger the error handler and cause program termination.
+  * In the future if we expose a builder-pattern method of creating producers, we could instead have the builder return an error indicating the segment is not mapped.
+  * An alternative to the above is to add an additional enum with configuration options allowing for the runtime to map requested segments on the fly. This increases flexibility, but because it is already possible to tell the runtime to map a specific segment, this would not add much value. A user who wishes to ensure a producer's segment is already mapped may simply call `mapSegmentForWriting(name)`, ignoring the error indicating that the segment has already been mapped.
+* When a consumer is created, an error will only occur if **no** segment has been mapped. Otherwise if a producer on the same channel as the consumer uses a segment the consumer does not have access to (whether because the segment is unmapped or because the consumer does not have read access), this will result in a fatal error. 
```

Comment:
Yeah I'll see if I can rephrase but what I mean is:

When you create a subscriber, you don't bother checking whether that subscriber will have appropriate access to what publishers publish. You just hope it does, and, as discussed in other threads, either capture an error or do some dynamic mapping on the fly of that segment (still checking for the error case that the subscriber just doesn't have read access).

The check I'm talking about here is just whether the subscriber has access to any segment at all, in the case where no dynamic mapping is allowed. This would mean that any `take()` will always fail and it would make sense to report this error early. 

For publishers on the other hand, you can more specifically check whether they have access to the segment they specify.